glean_parser/templates/javascript_server.jinja2

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// AUTOGENERATED BY glean_parser v{{ parser_version }}. DO NOT EDIT. DO NOT COMMIT.
+
+// This requires `uuid` and `mozlog` libraries to be in the environment
+const uuidv4 = require('uuid').v4;
+const mozlog = require('mozlog');
+
+const GLEAN_EVENT_MOZLOG_TYPE = 'glean-server-event';
+
+var _logger;
+
+{% for ping, metrics_by_type in pings.items() %}
+class {{ ping|event_class_name }} {
+ /**
+   * Create {{ ping|event_class_name }} instance.
+   *
+   * @param {string} applicationId - The application ID.
+   * @param {string} appDisplayVersion - The application display version.
+   * @param {string} channel - The channel.
+   */
+  constructor(applicationId, appDisplayVersion, channel, logger_options) {
+    this._applicationId = applicationId;
+    this._appDisplayVersion = appDisplayVersion;
+    this._channel = channel;
+
+    if (!_logger) {
+      // append '-glean' to `logger_options.app` to avoid collision with other loggers and double logging
+      logger_options.app = logger_options.app + '-glean';
+      // set the format to `heka` so messages are properly ingested and decoded
+      logger_options.fmt = 'heka';
+      _logger = mozlog(logger_options)();
+    }
+    this._log = _logger;
+  }
+  /**
+   * Record and submit a server event object.
+   * Event is logged using internal mozlog logger.
+   *
+   {% for metric_type, metrics in metrics_by_type.items() %}
+   {% for metric in metrics %}
+   * @param { {{-metric|js_metric_type-}} } {{ metric|metric_argument_name }} - {{ metric|metric_argument_description }}.
+   {% endfor %}
+   {% endfor %}
+   */
+  record({
+    {% for metric_type, metrics in metrics_by_type.items() %}
+    {% for metric in metrics %}
+    {{ metric|metric_argument_name }},
+    {% endfor %}
+    {% endfor %}
+  }) {
+    let timestamp = new Date().toISOString();
+    let eventPayload = {
+      metrics: {
+        {% for metric_type, metrics in metrics_by_type.items() %}
+        {{ metric_type }}: {
+          {% for metric in metrics %}
+          '{{ metric|metric_name }}': {{ metric|metric_argument_name }},
+          {% endfor %}
+        },
+        {% endfor %}
+      },
+      ping_info: {
+        seq: 0, // this is required, however doesn't seem to be useful in server context
+        start_time: timestamp,
+        end_time: timestamp,
+      },
+      // `Unknown` fields below are required in the Glean schema, however they are not useful in server context
+      client_info: {
+        telemetry_sdk_build: 'glean_parser v{{ parser_version }}',
```

Comment:
Feels a bit weird to use the glean_parser here as the SDK version, but it's also not ... wrong. Hmmm

Reply:
Yeah, after talking with Travis the other day I'm not blocking on this really. It's correct; and if in the future we determine this should report something different or live elsewhere everything here is self-contained enough to pull it off easily